### DISCUSSION TITLE
Howto programmatically control the steps component

### DISCUSSION THREAD

**Discussion by cmcclung-ANGENG** [ToolJet/ToolJet#7278] [Q&A] [answered]
Hi,

I'm looking to progress the steps id at an interval.

I've tried creating a JS query with an interval doing

components.steps1.currentStepId++;
alert(components.steps1.currentStepId);

The alert shows the updated count but the control doesn't show the change.

Is there a way to tell a component to redraw?

**Comment by tejakummarikuntla** [comment]
Hey @cmcclung-ANGENG ,
Can you try creating a Run JavaScript query with code 
`return components.steps1.currentStepId+1 ` and set the component's `current step` value to `{{queries.YOUR_RUN_JS_QUERY_NAME.data}}` and trigger the query on any events like button clicks?

Let me know if you need further help on this!

**Comment by cmcclung-ANGENG** [comment, marked as answer]
Hi,

I got it to to work.

I ended up using variables.

so the component looks to the variable the query updates and the js with the interval is in a seperate query.

e.g
steps component current step = {{variables.stepsNumber}}

JS query:
variables.stepsNumber++;
return variables.stepsNumber;

Then inside another JS query https://docs.tooljet.com/docs/how-to/run-query-at-specified-intervals/

and run that JS query created above when required.
 
I found that the clear interval had to happen in another query though.
